src/main/java/org/opentripplanner/middleware/persistence/TypedPersistence.java

```diff
@@ -86,8 +87,17 @@ public T create(String updateJson) {
      * TODO maybe merge this with the other create implementation above, passing in the base object and the updates.
      */
     public void create(T newObject) {
-        // TODO What happens if an object already exists with the same ID?
-        mongoCollection.insertOne(newObject);
+        try {
+            // TODO What happens if an object already exists with the same ID?
+            mongoCollection.insertOne(newObject);
+        } catch (Exception e) {
+            String objName = (newObject == null) ? "N/A" : newObject.getClass().getSimpleName();
+            String objContent = (newObject == null) ? "N/A" : newObject.toString();
+            String message = String.format("Unable to create object: %s with content: %s", objName, objContent);
+            LOG.error(message, e);
+            BugsnagReporter.reportErrorToBugsnag("Creating object", message, e);
+            throw e;
```

Comment:
I think this recent refactor of `reportErrorToBugsnag` perhaps should contain some of this logic. Instead of its signature being `String context, String message, Throwable e`, I think it would be more helpful to swap message for a generic `Object badEntity`. I think the easiest approach would be for me to just make a PR to this to demonstrate what I'm thinking.

Reply:
Merged #44